packages/service-library/src/servicelib/redis.py

```diff
@@ -72,11 +121,18 @@ async def _auto_extend_lock(lock: Lock) -> None:
             ttl_lock = self._client.lock(
                 lock_key, timeout=_DEFAULT_LOCK_TTL.total_seconds()
             )
-            if not await ttl_lock.acquire(blocking=False, token=lock_value):
-                raise AlreadyLockedError(lock=ttl_lock)
+
+            if not await ttl_lock.acquire(
+                blocking=blocking, token=lock_value, blocking_timeout=blocking_timeout_s
+            ):
+                # NOTE: acquisition fails because
+                # 1. the lock was already acquired by some other entity
+                # 2. timeouts out while waiting for lock to be free (another entity holds the lock)
+                raise CouldNotAcquireLockError(lock=ttl_lock)
```

Comment:
By your description, I do not see a difference between 1. and 2. In both cases, some other entity has the lock, right? 

Q: out of curiosity: what is the difference between ``CouldNotAcquireLockError``  and ``AlreadyLockedError``